### PR TITLE
Simplify React Component and Fix Imports

### LINKED ISSUE #4172: [Table] Add data test subjects to table components for easier testing
Some of the components for the table visualizations can add data test subjects to find them easier. For example: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/vis_type_table/public/components/table_vis_component_group.tsx#L25

Can be transform

```
        {tableGroups.map(({ table, title }, index) => (
          <div key={title} className="visTable__group" data-test-subj={`visTableGroup${index}`}>
            <TableVisComponent
              title={title}
              table={table}
              visConfig={visConfig}
              event={event}
              uiState={uiState}
            />
          </div>
        ))}
```

This will allow us to turn: https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/utils/dashboards/vis_type_table/table.js#L11

Into
```
cy.get(`[data-test-subj="visTableGroup${i}"]`)
```

This will help make the tests a little bit more stable and easier to read.

### PR DESCRIPTION
**Pull request by arpitjaswal**
### Description

This pull request updates the TableVisComponentGroup React component to improve test selectivity and enhance stability and readability. The changes include:

- Adding `data-test-subj` attributes to the visualization groups for improved test targeting.
- Removal of unused imports and variables.

These changes help ensure the reliability and maintainability of the component.

### Issues Resolved

Closes #4172

## Testing the changes

1. Run the following commands to ensure all tests pass:
   - `yarn test:jest`
   - `yarn test:jest_integration`
   - `yarn test:ftr`

2. Perform the following steps to validate the functionality of the changes:

   - Build and deploy the application with the changes.
   - Navigate to a page that uses the `TableVisComponentGroup` component.
   - Verify that the table visualizations are rendered correctly.
   - Interact with the table visualizations and ensure that sorting, filtering, and other actions work as expected.
   - Run the Cypress tests for the table visualizations to verify that the added `data-test-subj` attributes are correctly used for targeting elements.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/vis_type_table/public/components/table_vis_component_group.tsx [hunk 1]
```diff
@@ -1,11 +1,5 @@
-/*
- * Copyright OpenSearch Contributors
- * SPDX-License-Identifier: Apache-2.0
- */
```
Comment (BSFishy) [inline src/plugins/vis_type_table/public/components/table_vis_component_group.tsx:4]: This copyright header needs to stay in

File: src/plugins/vis_type_table/public/components/table_vis_component_group.tsx [hunk 2]
```diff
@@ -22,16 +16,16 @@ export const TableVisComponentGroup = memo(
   ({ tableGroups, visConfig, event, uiState }: TableVisGroupComponentProps) => {
     return (
       <>
-        {tableGroups.map(({ table, title }) => (
-          <EuiFlexItem key={title} className="visTable__group">
+        {tableGroups.map(({ table, title }, index) => (
+          <div key={title} className="visTable__group" data-test-subj={`visTableGroup${index}`}>
```
Comment (BSFishy) [inline src/plugins/vis_type_table/public/components/table_vis_component_group.tsx:20]: This element should remain `EuiFlexItem`. OUI (or EUI, as it is referred to in Dashboards) is our component library, and we want to make sure we're using it throughout the application. `EuiFlexItem` supports the `data-test-subj` prop, it should just be a matter of adding it.

**Review by BSFishy — COMMENTED** [review]
Thanks for the contribution! The original issue was unfortunately a little misguided. We want to keep our code using OUI (EUI), we just want to add the `data-test-subj` prop to this element. Additionally, the `CHANGELOG.md` needs to be updated with your change

**Comment by ananzh** [comment]
@arpitjaswal Thanks for working on this issue. Could you fix the comments and I think you could also help us to raise a PR to our functional test repo to fix https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/utils/dashboards/vis_type_table/table.js#L11 as mentioned in the issue.

**Comment by joshuarrrr** [comment]
@arpitjaswal I'm going to mark this as draft until you have time to revisit it.

**Comment by ananzh** [comment]
@arpitjaswal Due to an extended period without updates, we're going to close this issue for now. We appreciate your contribution and understand that priorities and availability can change. Please feel free to reopen this issue when you have the opportunity to revisit it. Thank you for your understanding.

**Pull request closed without merging**